corehq/apps/app_manager/templates/app_manager/v2/partials/module_view_settings.html

```diff
@@ -9,14 +9,15 @@
     <fieldset>
         {% block settings_fields %}
 
+
+          {% if child_module_enabled or module.module_type == 'shadow' or app.advanced_app_builder or module.put_in_root or app.grid_display_for_some_modules or value or ko_value or not module.is_surveys %}
```

Comment:
I think instead of `value or ko_value` you want to check for `module.module_filter`